File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
MAX_N = 10**6
isErased = [False] * (MAX_N + 1)


def sieve(N):
    isErased[0] = True
    isErased[1] = True
    for p in range(2, int(N ** 0.5) + 1):
        if isErased[p]:
            continue
        for j in range(p * p, N + 1, p):
            isErased[j] = True


m, n = map(int, input().split())
sieve(n)
for i in range(m, n + 1):
    if not isErased[i]:
        print(i)

File: 백준/Silver/1929. 소수 구하기/소수 구하기.swift
import Foundation

let input = readLine()!.split(separator: " ").map { Int($0)! }
let m = input[0], n = input[1]

if n < 2 { exit(0) }

var prime = [Bool](repeating: true, count: n + 1)
prime[1] = false

let limit = max(Int(sqrt(Double(n))), 2)

for i in 2...limit {
    if prime[i] {
        for j in stride(from: i * i, through: n, by: i) {
            prime[j] = false
        }
    }
}

for i in m...n {
    if prime[i] { print(i) }
}
